internal/app/newsletter/api/routers/routes.go

```diff
@@ -12,13 +13,19 @@ func Setup(app *fiber.App) {
 	newsletter := v1.Group("/newsletter")
 
 	subs := newsletter.Group("/subscribers")
-	// subs.Get("/", controllers.ListSubscribers)
-	subs.Post("/", controllers.AddSubscriber)
-	// subs.Get("/:id", controllers.GetSubscriber)
+
+	//unprotected routes
 	subs.Delete("/:id", controllers.DeleteSubscriber)
```

Comment:
We should protect this endpoint somehow. We can't protect it like the other routes, though. But we don't want anyone to be able to erase all our subscribers.

I am thinking of adding a secret and unique delete key for each subscriber, and only the requests that provide that key through a query param are allowed to delete the user.
| id | email         | delete_key |
|----|---------------|------------|
| 1  | sub@gmail.com | 281eea2    |
```
❌ DELETE {base-url}/v1/newsletter/subscribers/1
```
```
✅ DELETE {base-url}/v1/newsletter/subscribers/1?deleteKey=281eea2
```